### ISSUE TITLE
Thank you for awesome product. Scaling might be better.

### ISSUE THREAD

**Issue by fastengineer** [caprover/caprover#627]
I've used the following before. Heroku, AWS, K8S, Oracle, Dokku, Docker, Flynn, Rancher.
The result:

I love Heroku. Again and again I fell in love with Heroku. I have been using for 8 years.

About others are not comfortable. Except Rancher. Rancher is really amazing.

About Caprover: Thank you, it is really amazing. If scaling problems are solved, it can really be an alternative to Heroku.

Scaling problems I see in CapRover
A - All about database, I can not use a production database in CapRover seriously now.
1- Database sharding
2- Database replicating
3- Database backups
4- Database statistics
5- Database alerts

B- Different hardwares on servers. Limits are different for each cluster.
1- Be aware of limits and distribute containers in line with limits

C- Self Healing
1- If a cluster dies, purchase new cluster in various DC's from providers like DigitalOcean, Hetzner, Linode, ScaleWay, AWS, AliCloud, Google Cloud.
2- If all clusters hits limit create new cluster. 
3- If a container dies, create a new one.

D- Procfile
1 - We have different process types. Workers, Clocks, Web Apps. They scale differently. So there is need instance count setting for each process type described in Procfile.
2- Don't run different process types in same container

E- Statistics
1 - Like Heroku, only important statistics about web app. Request per Second, %95 Response Time, %99 Response Time.

F- Auto Scaling
1- Auto increment of container size. Rule based on [E]- Statistics
2- Auto increment of cluster size. Described in [C-1]

G- Global Services 
1 - Example. I can deploy a mail service with confidence. Like https://github.com/postalhq/postal
But I was feeling bored while attaching it to each application. Expected: I want to mark it as a service. I write service definition. Simply I click attach to app button, then I select the application. The ENV variable must be set in application.

H- Herokuish Deployment Section
1- I know there is different deployment methods. We have stable apps that lives in Heroku for years. As someone who has tried kubernetes and docker, I think Heroku is always more reliable.
I know I can deploy manually with herokuish but it is comfortable to have one click deploy. https://github.com/gliderlabs/herokuish

L- Logging
1- Name every container like Heroku Dyno Naming
1- It would be good to see all logs unified for each app.
2- Log filtering, for container, for process type, for keyword, for DateTime period.

S- Uploading Custom SSL
1- Like Heroku

X- External Service Attaching
1- When I want to use the same s3 account for several applications, should I enter all of them by line by line?
2- As I say in [G] section, attaching services operation can be exist for external services like S3.

Z- Exporting
1- Exporting app settings and metadata as json. App.json
2- Recreating it with exported json. App.json
3- You know this enables deploy to heroku button like features.

**_I can contribute to make all of them to work._**
**_I already developed enterprise grade products before._**
**_Only I need a starter guide to contribute._**

**Comment by githubsaturn** [comment]
Thanks for the comment and providing the suggestion. However, CapRover is meant to be a thin layer around Docker and nginx. From the [Contribution Guide](https://github.com/caprover/caprover/blob/master/CONTRIBUTING.md):
> CapRover is not an enterprise grade application like Kubernetes. Do not patch it with half-done features that make it look like one - it will eventually fail as we don't have resources to support such features.- CapRover scope is a helper around Docker, nginx and Let's Encrypt.

Lots of items that you mentioned are enterprise features that require lots of engineering work which is absolutely beyond scope of CapRover. For example, if you're able to do the items you mentioned under "Self Healing" and "Database sharding and backup", you have a cloud database provider that can scale easily and it will be a multi million dollar business :) For example, mLab was essentially a cloud database and was [acquired for $68m](https://venturebeat.com/2018/10/10/mongodb-to-acquire-cloud-database-provider-mlab-for-68-million/), worth noting that they "only" supported MongoDB.

To sum up, these features are definitely very nice to have, but the engineering effort to build a "non-hacky" solution for these is definitely beyond the scope of CapRover and more towards an enterprise solution. CapRover is tageting the simple apps with small scope. This small scope enabled CapRover to be stable and bug-free. Increasing the scope of CapRover will make this project unmaintainable.
